File: MajorityFrequencyCharacters.py
from collections import Counter, defaultdict
class Solution:
    def majorityFrequencyCharacters(self, s: str) -> str:
        fr = Counter(s)
        fr2 = defaultdict(list)

        for c, count in fr.items():
            fr2[count].append(c)
        
        best= -1
        ans = ""

        for k in sorted(fr2.keys(), key=lambda x: -x):

            if len(fr2[k]) > best:
                best = len(fr2[k])
                ans = "".join(fr2[k])
        return ans


solutionTest = Solution()
print(solutionTest.majorityFrequencyCharacters("abababacde"))  # Output: "abcde"

File: MajorityFrequencyCharacters.cpp
#include <algorithm>
#include <string>
#include <iostream>
#include <unordered_map>
#include <vector>
using namespace std;

class Solution {
   public:
    string majorityFrequencyCharacters(string caractInput) {
        unordered_map<char, int> mapFreq;
        vector<pair<int, char>> items;
        int best = -1;
        string ans;

        for (char c : caractInput) mapFreq[c]++;

        for (auto &p: mapFreq) {
            items.push_back(make_pair(p.second, p.first));
        }

        sort(items.begin(), items.end(), [](auto &a, auto &b) {
            return a.first < b.first;
        });

        unordered_map<int, string> freqGrupo;
        for (auto &p: items) {
            freqGrupo[p.first].push_back(p.second);
        }


        for (auto &entry: freqGrupo) {
            int coutChars = entry.second.size();

            if (coutChars > best) {
                best = coutChars;
                ans = entry.second;
            }
        }

        return ans;

    }
};

int main() {
    Solution solutionTest;
    cout << solutionTest.majorityFrequencyCharacters("abababacde") << endl;
    return 0;
}